bftengine/include/bftengine/ControlStateManager.hpp

```diff
@@ -0,0 +1,31 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+#include <string>
+#include <unordered_map>
+#include "IStateTransfer.hpp"
+
+namespace bftEngine {
+
+class ControlStateManager {
+ public:
+  void setStopAtNextCheckpoint();
+  bool getStopAtNextCheckpoint();
```

Comment:
Are we really just setting a flag and then getting whether the flag is set? That's what these names read like to me. I think it makes a bit more sense to indicate what they are actually doing. I'd prefer to rename them and provide some extra functionality.

`void stopAtNextCheckpoint(); ` actually calculates the checkpoint to stop at and writes that value into a reserved page.
`std::optional<uint64_t> getNextCheckpointToStopAt` returns the checkpoint number to stop at or `std::nullopt` if it's not set.

Reply:
done